documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -65,29 +81,14 @@ When used with the OAuth 2.0 over PLAIN callbacks, Kafka clients authenticate wi
 
 * A long-lived access token, obtained manually at configuration time
 
-The client must be enabled to use PLAIN authentication, and provide a `username` and `password`.
-If the password is prefixed with `$accessToken:` followed by the value of the access token, the Kafka broker will interpret the password as the access token.
+For both methods, the client must provide a `username` and `password` to pass credentials to the Kafka broker.
+If the password is prefixed with `$accessToken:` followed by the value of an access token, the Kafka broker will interpret the password as the access token.
 Otherwise, the Kafka broker will interpret the `username` as the client ID and the `password` as the client secret.
 
```

Comment:
```suggestion
When listener has no `tokenEndpointUri` configured, the `password` parameter is never interpreted as a `secret` since it can't be exchanged for an access token without a `tokenEndpoinyUri`. In this case `secret` is always interpreted as an unprefixed `access token`. 
```

Reply:
Not sure I understand what we're saying here. Is this to replace the preceding sentence, which seems to contradict it?

What we have so far:

> For both methods, the client must provide the PLAIN `username` and `password` properties to pass credentials to the Kafka broker.
> For the Kafka broker to interpret a password as an access token, the following is required:
> 
> * The listener configuration must have a token endpoint (`tokenEndpointUri`) configured 
> * The password requires an `$accessToken:` prefix followed by the value of the access token
> 
> Otherwise, the Kafka broker will interpret the `username` as the client ID and the `password` as the client secret.

Reply:
> If a token endpoint is not configured on the broker listener, ...

Doesn't that sound strange, to first make a categorical claim in the previous paragraph that two conditions are required to use access token for authentication, but in the next sentence add another case (negating that the two conditions are required) where we say that if the first condition is not fulfilled then it's also possible to use access token. Maybe someone else can chime in (@scholzj ?). I would prefer to start from server config being with tokenEndpointUri or without tokenEndpointUri and depending on that explain the options for credentials, rather than trying to explain how to pass access token which is the angle from which the explanation goes here. But maybe that's just me.
 
> the password should not include

Maybe use `the password parameter` or `the password property` to make it clear we have _that_ password in mind.

Reply:
Thnaks, @mstruk. Let's rearrange a little:

> For both methods, the client must provide the PLAIN `username` and `password` properties to pass credentials to the Kafka broker. The client uses these properties to pass a client ID and secret or username and access token.
> 
> Client IDs and secrets are used to obtain access tokens.
> 
> Access tokens are passed as `password` property values. 
> You pass the access token with or without an `$accessToken:` prefix. 
> 
> * If you configure a token endpoint (`tokenEndpointUri`) in the listener configuration, you need the prefix.
> * If you don't configure a token endpoint (`tokenEndpointUri`) in the listener configuration, you don't need the prefix. The Kafka broker interprets the password as a raw access token.

Reply:
Yeah I think that's better.